api/server/volume.go

```diff
@@ -1025,15 +1025,13 @@ func (vd *volAPI) snapEnumerate(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	snaps := make([]*api.Volume, len(resp.GetVolumeSnapshotIds()))
-	for i, s := range resp.GetVolumeSnapshotIds() {
+	snaps := make([]*api.Volume, 0)
+	for _, s := range resp.GetVolumeSnapshotIds() {
 		vol, err := volumes.Inspect(ctx, &api.SdkVolumeInspectRequest{VolumeId: s})
 		if err != nil {
-			e := fmt.Errorf("Failed to inspect volumeID: %s", err.Error())
-			vd.sendError(vd.name, method, w, e.Error(), http.StatusBadRequest)
-			return
+			continue
```

Comment:
Instead of blinding ignoring all errors, this should just ignore the volume not found error.

Reply:
Need to maintain the previous behavior where enumerate doesn't report errors for not found, but returns for other errors.